resources/gen-ai/aws-web-crawler-container/src/utils.ts

```diff
@@ -0,0 +1,71 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+export abstract class Utils {
+  static async wait(ms: number): Promise<void> {
+    return new Promise((resolve) => setTimeout(resolve, ms));
+  }
+
+  static getFileType(url: string) {
+    const fileRegex = /(?:\.([^.]+))?$/;
+    url = url.split('/').pop() ?? '';
+    const check = fileRegex.exec(url);
+    let extension = check && check.length > 1 ? check[1] : undefined;
+    extension = extension?.split('?')[0].split('#')[0];
+
+    return extension ? extension.toLowerCase() : undefined;
+  }
+
+  static getBaseUrl(url: string) {
+    url = url.trim();
+    if (!url.toLowerCase().startsWith('http://') && !url.toLowerCase().startsWith('https://')) {
+      url = `https://${url}`;
+    }
+
+    try {
+      const parsedUrl = new URL(url);
+      return `${parsedUrl.protocol}//${parsedUrl.hostname}`;
+    } catch (error) {
+      return undefined;
+    }
+  }
+
+  static unique<T>(array: T[]) {
+    return Array.from(new Set(array));
+  }
+
+  static generateDateTimeStringWithMilliseconds() {
+    const formatNumber = (num: number, digits: number) => {
+      return num.toLocaleString('en-US', {
+        minimumIntegerDigits: digits,
+        useGrouping: false,
+      });
+    };
+
+    const now = new Date();
+    const year = formatNumber(now.getFullYear(), 4);
+    const month = formatNumber(now.getMonth() + 1, 2);
+    const day = formatNumber(now.getDate(), 2);
+    const hours = formatNumber(now.getHours(), 2);
+    const minutes = formatNumber(now.getMinutes(), 2);
+    const seconds = formatNumber(now.getSeconds(), 2);
+    const milliseconds = formatNumber(now.getMilliseconds(), 3);
+
+    const retValue = `${year}-${month}-${day}-${hours}-${minutes}-${seconds}${milliseconds}`;
```

Comment:
```suggestion
    const now = new Date();
    const retValue = `${now.toISOString()}`;
```

Reply:
ToISOString > "2024-06-10T18:44:21.978Z"
Existing code > "2024-06-10-13-44-21978"

Reply:
Fixed in [475de44](https://github.com/awslabs/generative-ai-cdk-constructs/pull/474/commits/475de44a683aed19f7e07033e48bd82972e099eb)